### PR TITLE
docs: update lifecycle specifications

### PR DESCRIPTION
**Pull request by robszumski**
- use consistent formatting
 - add suggested workloads

### CODE REVIEW

File: Documentation/platform-lifecycle.md [hunk 1]
```diff
@@ -20,19 +20,25 @@ Development workflows only
 ### Alpha
 
 - Manually tested and can reliably produce minimally functioning clusters
-  - Kuberntes API works
-  - Authenticated Tectonic Console works
+- Kuberntes API works
```
Comment (zbwright) [inline Documentation/platform-lifecycle.md]: Kubernetes

Comment (zbwright) [inline Documentation/platform-lifecycle.md]: also - above this line, 'underging'  should be 'undergoing' (I can't figure out how to add comments to unedited lines.)

File: Documentation/platform-lifecycle.md [hunk 2]
```diff
@@ -54,27 +60,43 @@ Development workflows only
...
 - Assets are packaged into the official Tectonic Installer tarball with every Tectonic release.
+- Documentation is published to Tectonic docs on coreos.com
 - (Optional) Installer UI is built to guide user through the configuration process.
 
 *User Flow*
 
-1. Use GUI installer to generate `terraform.tfvars` file, or manually create it.
+GUI Only
+
+1. Use GUI installer configure cluster
```
Comment (zbwright) [inline Documentation/platform-lifecycle.md]: Use GUI installer to configure cluster.

question: *Packaging* list says the GUI installer is optional, but the user flow calls out the GUI installer. which is it for beta?

File: Documentation/platform-lifecycle.md [hunk 3]
```diff
@@ -54,27 +60,43 @@ Development workflows only
...
+Non-GUI
+
+1. Manually create `terraform.tfvars` file
 1. Use static Terraform assets contained in installer tarball
 1. Manually run terraform
 
+*Suitable Workloads*
+
+- Testing in pre-production environments
+- Not recommended production
```
Comment (zbwright) [inline Documentation/platform-lifecycle.md]: Not recommended for production

File: Documentation/platform-lifecycle.md [hunk 4]
```diff
@@ -83,10 +105,14 @@ GUI Only
 1. Use GUI installer configure cluster
```
Comment (zbwright) [inline Documentation/platform-lifecycle.md]: same as above.

**Comment by robszumski** [comment]
Updated based on comments. I must have really been fried, there were a lot of words missing 😓

**Pull request closed without merging**